lib/postprocessing/image.js

```diff
@@ -94,113 +93,50 @@ function processBasicImageOperations(img, instructions) {
 
-    return new Promise(function (resolve, reject) {
-        const outdir = path.dirname(outfile);
-        const renditionName = path.basename(outfile);
-
-        // for pdfs and multi-page docs/images, take the first one only
-        let img = gm(`${infile}[0]`);
-
-        // const selectedOperations = operations || DEFAULT_OPERATIONS;
-        // turning off watermarking for increamental post-processing
-        const selectedOperations = DEFAULT_OPERATIONS;
-        img = applyImageOperations(img, instructions, selectedOperations);
-
-        img.write(outfile,
-            function (err) {
-                if (err) {
-                    console.log("FAILURE of rendition processing using image processing lib (imagemagick)", renditionName);
-                    console.error("failed conversion:", err);
-                    reject(`conversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                } else {
-                    // If we specified jpegSize for JPEG output we need to verify that the size does not exceed that
-                    const size = fileSize(outfile);
-                    if (instructions.fmt === "jpg" && instructions.jpegSize && size > instructions.jpegSize) {
-                        const newImg = gm(outfile);
-
-                        // From http://www.imagemagick.org/Usage/formats/#jpg
-                        // As of IM v6.5.8-2 you can specify a maximum output filesize for the JPEG image.
-                        newImg.define(`jpeg:extent=${instructions.jpegSize}`);
-                        const tmpFile = `${outdir}/${tempFileName("jpg")}`;
-                        newImg.write(tmpFile,
-                            function (err) {
-                                if (err) {
-                                    console.log("FAILURE of rendition processing (fmt=jpg) using image processing lib (imagemagick)", renditionName);
-                                    console.error("failed conversion:", err);
-                                    reject(`cconversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                                } else {
-                                    fs.renameSync(tmpFile, outfile);
-                                    console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                                    resolve(true);
-                                }
-                            });
-                    } else {
-                        console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                        resolve(true);
-                    }
-                }
-            });
-    });
+    const img = await read(infile);
+
+    await render(img, instructions);
+
+    await write(img, outfile, instructions);
```

Comment:
Would also be nicer, I think, to be able to chain the calls (fluent API mentionned above).